pkg/mcs/scheduling/server/rule/watcher.go

```diff
@@ -175,3 +254,38 @@ func (rw *Watcher) Close() {
 	rw.cancel()
 	rw.wg.Wait()
 }
+
+// SetClusterComponents sets the cluster components for the watcher.
+func (rw *Watcher) SetClusterComponents(
+	sc *checker.Controller,
+	rm *placement.RuleManager,
+	rl *labeler.RegionLabeler,
+) {
+	rw.checkerController.Store(sc)
+	rw.ruleManager.Store(rm)
+	rw.regionLabeler.Store(rl)
+}
```

Comment:
We may need to reload once after setting to make sure we don't lose any put/delete events happening between the gaps.